package.json

```
@@ -22,6 +22,10 @@
   },
   "license": "MIT",
   "author": "stylelint",
+  "exports": {
+    "import": "./lib/index.mjs",
+    "require": "./lib/index.cjs"
+  },
```

Export also `package.json`:

```suggestion
  "exports": {
    ".": {
      "import": "./lib/index.mjs",
      "require": "./lib/index.cjs"
    },
    "./package.json": "./package.json"
  },
```

I had a problem with a package that didn't export `package.json`: https://github.com/depcheck/depcheck/issues/754#issuecomment-1582636531

I haven't found any official documentation about this, but other packages export `package.json`: [yargs](https://github.com/yargs/yargs/pull/1818), [typedoc](https://github.com/TypeStrong/typedoc/commit/e021751e5fb599b944994e6c2225349a88b51caa#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR18), [eslint](https://github.com/eslint/eslint/blob/926a286/package.json#L11)...

---

This is already covered by #6258.
i.e. these migration PRs have to avoid introducing regressions

---

Right. I've reverted the change via 80a1ef014.

---

@regseb Thanks for the info. Commented on https://github.com/stylelint/stylelint/issues/6258#issuecomment-1710035791